widget/ui/src/components/StatusDrawer/StatusDrawer.tsx

```diff
@@ -1,82 +1,126 @@
-            {status === 'success' ? (
-              <CheckCircleIcon color="success" size={24} />
-            ) : status === 'failed' ? (
-              <InfoCircleIcon color="error" size={24} />
-            ) : (
-              <WarningIcon color="warning" size={24} />
-            )}
-            <Typography variant="title" size="small">
-              {title}
-            </Typography>
-          </Header>
-          <Typography variant="body" size="small">
-            {subtitle}
-          </Typography>
+          {Array.isArray(conditions) ? (
+            <div>
+              {conditions.map((condition, index) => (
+                <>
+                  {index !== ZERO && <Divider size={20} />}
+                  <Accordion
+                    key={condition.id}
+                    content={
+                      condition.details && (
+                        <Content
+                          details={condition.details}
+                          onClick={condition.onClick}
+                        />
+                      )
+                    }>
+                    <StatusCard
+                      title={condition.title}
+                      description={condition.description}
+                      type={condition.status}
+                    />
+                  </Accordion>
+                </>
+              ))}
+            </div>
+          ) : (
+            <StatusCard
+              title={conditions.title}
+              description={conditions.description}
+              type={conditions.status}>
+              {conditions.details && (
+                <Content
+                  details={conditions.details}
+                  onClick={conditions.onClick}
+                />
+              )}
+            </StatusCard>
+          )}
         </>
       }
       footer={
-        <Footer>
-          {status !== 'failed' && (
-            <Button fullWidth onClick={onCancel}>
-              {status === 'success' ? 'Cancel' : 'See detailes'}
-            </Button>
-          )}
-          <Divider size={16} direction="horizontal" />
-          <Button
-            fullWidth
-            onClick={onSubmit}
-            variant="contained"
-            type="primary">
-            {status === 'success'
-              ? 'Done'
-              : status === 'continue'
-              ? 'Continue'
-              : 'See detailes'}
-          </Button>
-        </Footer>
+        (!!onConfirm || !!onSecondaryButton) && (
+          <Footer>
+            {!!onConfirm && (
+              <Button fullWidth type="primary" variant="contained">
+                {confirmTitle}
+              </Button>
+            )}
+            {!!onSecondaryButton && (
+              <>
+                <Divider size={12} />
+                <Button fullWidth type="primary" variant="outlined">
+                  {secondaryButtonTitle}
+                </Button>
+              </>
+            )}
+          </Footer>
+        )
```

Comment:
I think you can dynamically get the footer as a prop and remove hardcoded values (like type, variant, ...) for these buttons.